utils/docker/vcluster/README.md

```diff
@@ -0,0 +1,199 @@
+    Some distributions are not yet well supported such as
+    [rocky8.5](https://hub.docker.com/r/rockylinux/rockylinux): issue with the management of
+    hugepages with the [spdk](https://spdk.io/) library.
+
+
+## Building Docker Images
+
+### Building Base DAOS Image
+
+The first image to create is the `daos-base` image which is not intetended to be used as it, but as
+a base image for building the other three daos images.  This first image could be built directly
+from GitHub with the following command:
+
+```bash
+$ docker build --tag daos-base:centos8 \
+	https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/daos-base/el8
+```
+
+or from a local tree with the following command:
+
+```bash
+$ docker build --tag daos-base:centos8 utils/docker/vcluster/daos-base/el8
+```
+
+This Docker file accept the following arguments:
+
+- `RHEL_BASE_IMAGE`: Base docker image to use (default centos)
+- `RHEL_BASE_VERSION`: Version of the base docker image to use (default 8)
+- `BUST_CACHE`: Manage docker building cache (default undefined).  To invalidate the cache, a random
+	value such as the date of the day shall be given.
+- `DAOS_AUTH`: Enable DAOS authentication when set to "yes" (default "no")
+- `DAOS_REPOS`: Space separated list of repos needed to install DAOS (default
+	"https://packages.daos.io/v2.0")
+- `DAOS_GPG_KEYS`: Space separated list of GPG keys associated with DAOS repos (default
+	"https://packages.daos.io/RPM-GPG-KEY")
+- `DAOS_REPOS_NOAUTH`: Space separated list of repos to use without GPG authentication
+	(default "")
+
+For example, building a DAOS base image with authentication enabled could be done with the
+following command:
+
+```bash
+$ docker build --tag daos-base:centos8 --build-arg DAOS_AUTH=yes \
+	https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/daos-base/el8
+```
+
+### Building DAOS Nodes Images
+
+The three images `daos-server`, `daos-admin` and `daos-client` could be built directly from GitHub
+or from a local tree in the same way as for the `daos-base` image.  Following command could be used
+to build directly the three images from GitHub:
+
+```bash
+$ for image in daos-server daos-admin daos-client ; do \
+	docker build --tag "$image:centos8" \
+		"https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/$image/el8"; \
+  done
+```
+
+The Docker file of the `daos-server` image accept the following arguments:
+
+- `DAOS_BASE_IMAGE`: Base docker image to use (default daos-base)
+- `DAOS_BASE_VERSION`: Version of the base docker image to use (default centos8)
+- `DAOS_AUTH`: Enable DAOS authentication when set to "yes" (default "no")
+- `DAOS_HUGEPAGES_NBR`: Number of huge pages to allocate for SPDK (default 4096)
+- `DAOS_IFACE_NAME`: Fabric network interface used by the DAOS engine (default "eth0")
+- `DAOS_SCM_SIZE`: Size in GB of the RAM emulating SCM devices (default 4)
+- `DAOS_BDEV_SIZE`: Size in GB of the file created to emulate NVMe devices (derault 16)
```

Comment:
It would be handy here to have a note that the IP of this interface will be moved later, perhaps re-order these so you can have more text for DAOS_IFACE_NAME without losing the visual fact that these are a block.

Minor, but "lo" might be preferably here to "eth0".

You also have a spelling mistake here.  I'm not verified the marking rending of this proposed change.
```suggestion
- `DAOS_SCM_SIZE`: Size in GB of the RAM emulating SCM devices (default 4)
- `DAOS_BDEV_SIZE`: Size in GB of the file created to emulate NVMe devices (default 16)
- `DAOS_IFACE_NAME`: Fabric network interface used by the DAOS engine (default "eth0")
  The IP address for this interface will be required when starting DAOS.
```

Reply:
I have some issue with the loopback interface at the beginning.
I am going to test it again: indeed it will be more portable.

Reply:
Yes, I have some issue with hwloc and the loopback interface (as for the docker bridge network).
I am going to add a statement on this point in the limitation warning.
@johannlombardi is aware of this limitation with hwloc. It is planned to fix it later. Please @johannlombardi , could you confirm this last point.

Reply:
> It would be handy here to have a note that the IP of this interface will be moved later, perhaps re-order these so you can have more text for DAOS_IFACE_NAME without losing the visual fact that these are a block.
> 
> Minor, but "lo" might be preferably here to "eth0".
> 
> You also have a spelling mistake here. I'm not verified the marking rending of this proposed change.

Going to integrate your remark for the interface. However, do not find the spelling mistake.

Reply:
"derault"

Reply:
Arghhh, thanks

Reply:
fixed with commits:
- https://github.com/daos-stack/daos/pull/7556/commits/83667f55edd452cc0e00bfd93c2952c46ef4a951
- https://github.com/daos-stack/daos/pull/7556/commits/f1b52574e79de789a736911b401f0b55e424915f

Reply:
> Sadly, I have some issue with hwloc and the loopback interface (as for the docker bridge network). I am going to add a statement on this point in the limitation warning. @johannlombardi is aware of this limitation with hwloc. It is planned to fix it later. Please @johannlombardi , could you confirm this last point.

Add comment on loopback support issue in commit https://github.com/daos-stack/daos/pull/7556/commits/9972f9cab1b697b03b5ebda59a6d89ee03f90c7b